en/docs/guides/adaptive-auth/typingdna-based-adaptive-auth.md

```diff
@@ -0,0 +1,119 @@
+
+You can integrate typingDNA with WSO2 Identity Server to provide risk-based adaptive authentication for users.
+
+----
+
+## Scenario
+Consider a scenario where you want to prompt an additional authentication step if the typing pattern of the user trying to log in does not match the typing pattern registered in the user's account. Then the log in flow of the user should be stepped up as follows:
+
+1. Basic authentication (username and password)
+2. TOTP
+
+----
+
+## Prerequisites
+1. You need to [set up the sample]({{base_path}}/guides/adaptive-auth/adaptive-auth-overview/#set-up-the-sample)   application.
+2. [Create a user]({{base_path}}/guides/identity-lifecycles/admin-creation-workflow/) named **Alex** with login permissions.
+3. Create a typingDNA account. Learn how to create one [here](https://github.com/wso2-extensions/identity-conditional-auth-typingdna/blob/main/docs/files/Account%20Creation.pdf).
+
+    !!! info
+        Once you sign up for a typingDNA account, go to the dashboard and under API settings, enable **Auto-enroll**   and **Force initial enrollments**.
+
+## Set up TypingDNA in WSO2 IS
+
+Follow the guide below to set up typingDNA in the WSO2 IS server
+
+1. Open the **deployment.toml** file found in the **{IS-Home}/repository/conf/** directory and add the following configuration.
+	```
+	[myaccount.security]
+    enabled_features=["security.loginVerifyData.typingDNA"]
+    ```
+
+2. Go to the [WSO2 store](https://store.wso2.com/store/assets/isconnector/list) and download both the Authenticator and the Artifacts from the **TypingDNA Connector**.
+
+3. Copy the Authenticator file, `org.wso2.carbon.identity.conditional.auth.typingdna.functions-x.x.x.jar` and paste it in the **{IS-Home}/repository/components/dropins** directory.
+
+4. Unzip the Artifacts archive, copy the `api#identity#typingdna#v_.war` file and paste it in the **{IS-Home}/repository/deployment/server/webapps** directory.
+
+5. Restart the WSO2 IS.
+
+6. Navigate to **Identity Providers -> Resident -> Other Settings -> TypingDNA Configuration** and make the following changes.
+
+    - Enable TypingDNA
+    - Configure the typingDNA API key and API secret retrieved from the typingDNA [dashboard](https://www.typingdna.com/clients/).
+    - Enable Advance TypingDNA-API mode if you have a pro/enterprise typingDNA account.
+    - Configure the region (eu or us).
+
+    ![TypingDNA configuration]({{base_path}}/assets/img/samples/typingdna-configuration.png)
+
+7. Click **Update** to save the changes.
+
+## Configure TypingDNA in applications
+
+Follow the steps below to configure TypingDNA in your application.
+
+1. Navigate to **Service Providers -> List** and click **Edit** on the service provider that you want to configure TypingDNA in.
+
+2. Expand **Local and Outbound Authentication Configuration** and click **Advanced Configuration**.
+3. Configure two Authentication steps.
```

Comment:
```suggestion
3. Configure the two authentication steps.
```

Reply:
It could be any authentication steps. Basic and TOTP were used only for the example. I think it could be left as the original.